Index: src/main/java/com/tngtech/java/junit/dataprovider/internal/convert/ObjectArrayConverter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/tngtech/java/junit/dataprovider/internal/convert/ObjectArrayConverter.java	(revision edd22cb2bfd8d27cc34c163af68fd18fcf517b62)
+++ src/main/java/com/tngtech/java/junit/dataprovider/internal/convert/ObjectArrayConverter.java	(revision )
@@ -34,9 +34,10 @@
 
     private Object convertVarArgArgument(Object[] data, Class<?> varArgComponentType, int nonVarArgParameters) {
         if (data.length > 0) {
-            Class<?> lastArgType = data[data.length - 1].getClass();
-            if (lastArgType.isArray() && lastArgType.getComponentType() == varArgComponentType) {
-                return data[data.length - 1];
+            Object date = data[data.length - 1];
+            Class<?> lastArgType = date != null ? date.getClass() : null;
+            if (lastArgType != null && lastArgType.isArray() && lastArgType.getComponentType() == varArgComponentType) {
+                return date;
             }
         }
 
Index: src/test/java/com/tngtech/java/junit/dataprovider/internal/convert/ObjectArrayConverterTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/com/tngtech/java/junit/dataprovider/internal/convert/ObjectArrayConverterTest.java	(revision edd22cb2bfd8d27cc34c163af68fd18fcf517b62)
+++ src/test/java/com/tngtech/java/junit/dataprovider/internal/convert/ObjectArrayConverterTest.java	(revision )
@@ -42,6 +42,19 @@
     }
 
     @Test
+    public void testConvertShouldHandleNullElementAsSingleElement() throws Exception {
+        // Given:
+        Object[] data = new Object[] { null };
+        Class<?>[] parameterTypes = new Class<?>[] { String.class, int[].class };
+
+        // When:
+        Object[] result = underTest.convert(data, true, parameterTypes);
+
+        // Then:
+        assertThat(result).containsExactly(null, new int[0]);
+    }
+
+    @Test
     public void testConvertShouldCreateEmptyVarargsArrayForLastMissingVarargsArgument() {
         // Given:
         Object[] data = new Object[] { "test" };
